### PR TITLE
Ik tables

### PR DESCRIPTION
**Pull request by irina0606**
https://trello.com/c/N6jspm4l/947-admin-tablescreate-embedded-table-at
Admin> Tables>Create Embedded Table: TC1

### CODE REVIEW

**Review by kgold2018 — APPROVED** [review]
ok

**Review by 4stuntman4 — APPROVED** [review]

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 1]
```diff
@@ -105,6 +105,11 @@ public ConfigurationPage(WebDriver driver) {
         super(driver);
     }
 
+    public MainPage clickHomeButton() {
+        homeButton.click();
+        return new MainPage(getDriver());
+    }
```
Comment (oooleks) [inline src/test/java/model/admin/common/ConfigurationPage.java:111]: We have this method already at Main page (goHomePage()), hence it should be deleted from here

File: src/test/java/test/admin/AdminTablesTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                .clickCreateField();
+
+        new AdminCreateFieldPage(getDriver())
+                .createField(LABEL_Field2, TEXT);
+
+        new AdminFieldsPage(getDriver())
+                .clickCreateField();
+
+        new AdminCreateFieldPage(getDriver())
+                .createField(LABEL_Field3, INT);
```
Comment (oooleks) [inline src/test/java/test/admin/AdminTablesTest.java:59]: Use method `createField(WebDriver driver, String label, String fieldType)` instead from ProjectUtils.Portal

File: src/test/java/test/admin/AdminTablesTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        new AdminEmbeddedEntitiesPage(getDriver())
+                .createEmbeddedEntity(EMB_LABEL1)
+                .clickEntities()
+                .clickConfigEntity(LABEL_Entity)
+                .clickTables()
+                .createEmbeddedEntity(EMB_LABEL2);
+
+        new ConfigurationPage(getDriver())
+                .clickHomeButton();
```
Comment (oooleks) [inline src/test/java/test/admin/AdminTablesTest.java:72]: Use existed goHomePage() method instead.

File: src/test/java/test/admin/AdminTablesTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        new MainPage(getDriver())
+                .clickEntity(LABEL_Entity);
+    }
```
Comment (oooleks) [inline src/test/java/test/admin/AdminTablesTest.java:76]: Is test finished? There is no any validation.

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 1 above)
Comment (irina0606) [reply src/test/java/model/admin/common/ConfigurationPage.java]: Done

File: src/test/java/test/admin/AdminTablesTest.java (hunk 2 above)
Comment (irina0606) [reply src/test/java/test/admin/AdminTablesTest.java]: not used any more

File: src/test/java/test/admin/AdminTablesTest.java (hunk 3 above)
Comment (irina0606) [reply src/test/java/test/admin/AdminTablesTest.java]: done

File: src/test/java/test/admin/AdminTablesTest.java (hunk 4 above)
Comment (irina0606) [reply src/test/java/test/admin/AdminTablesTest.java:61]: Yes, sir

File: src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        createButton.click();
+        waitForCmdHistory(name);
+        return new AdminFieldsPage(getDriver());
+    }
+
+    public String getHyperlink(WebElement name){
+       return name.getText();
+    }
+
+    public WebElement getHyperlinkTeachers() {
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java:38]: Do not call parameters in POM using any test-data: "Teachers", "Students". Be generic in naming.

File: src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package model.admin.configuration.entities;
+
+import model.admin.common.ConfigurationPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import utils.TestUtils;
+
+public class AdminEmbeddedEntitiesPage extends ConfigurationPage {
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java:9]: Why not to use AdminTabelsPage which you created already below for the same?

File: src/test/java/test/admin/AdminTablesTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        new AdminEmbeddedEntitiesPage(getDriver())
+                .clickConfigEntity(LABEL_Entity)
+                .clickTables();
+
+        new AdminEmbeddedEntitiesPage(getDriver())
+                .createEmbeddedEntity(EMB_LABEL2);
+
+        new AdminEmbeddedEntitiesPage(getDriver())
+                .clickConfigEntity(LABEL_Entity)
+                .clickTables();
```
Comment (oooleks) [inline src/test/java/test/admin/AdminTablesTest.java:55]: Could by simplified without creating so many AdminEmbeddedEntitiesPage objects

File: src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+    @FindBy(xpath = "//input[@id='new-emb-entity-label']")
+    private WebElement embeddedLabelName;
+
+    @FindBy(xpath = "//button [@onclick='createEmbedded(this)']")
+    private WebElement createButton;
+
+    @FindBy(xpath = "//div[@class='content']//a[1]")
+    private WebElement hyperlinkTeachers;
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java:18]: Do not use test-data when naming elements

File: src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    @FindBy(xpath = "//div[@class='content']//a[2]")
+    private WebElement hyperlinkStudents;
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminEmbeddedEntitiesPage.java:21]: Do not use test-data when naming elements

**Review by guram21 — APPROVED** [review]

**Pull request merged**